src/main/java/io/pinecone/PineconeIndexOperationClient.java

```diff
@@ -0,0 +1,31 @@
+package io.pinecone;
+
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+
+import java.io.IOException;
+
+public class PineconeIndexOperationClient {
+
+    private AsyncHttpClient client;
+
+    PineconeIndexOperationClient(AsyncHttpClient client) {
+        this.client = client;
+    }
+
+    public void deleteIndex(String environment, String apiKey, String indexName) throws IOException {
+        String url = "https://controller." + environment + ".pinecone.io/databases/" + indexName;
+        System.out.println("Sending delete index request:");
+        // ToDo: Capture Response response and customized error messages
+        client.prepare("DELETE", url)
```

Comment:
This is a blocking call; if we want to mimic Python client behavior, we should add a timeout. The default is 300s and the user can pass a custom timeout, with -1 being returned immediately. Not needed immediately though I suppose

Reply:
Good point!